platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileReader.java

```diff
@@ -66,15 +66,19 @@ public static List<SavedStateInfo> getSavedStateFiles(
     /**
      * Reads a SignedState from disk
      *
-     * @param platformContext the platform context
-     * @param stateFile
-     * 		the file to read from
+     * @param platformContext               the platform context
+     * @param stateFile                     the file to read from
+     * @param deleteStateOnBackgroundThread if true, the state file will be deleted on a background thread. If false,
```

Comment:
Curious - what consideration informs whether a state file will be deleted on a background thread vs other?

Reply:
This is a feature that is useful in unit testing. In a unit test this should always be `false` unless the unit test has also wired up a state garbage collector. In production, this should always be `true`.

Reply:
thank you. I think there should be a documentation to that effect.

Reply:
Expanded the javadoc.

```
     * @param deleteStateOnBackgroundThread if true, the state file will be deleted on a background thread. If false,
     *                                      the state file will be deleted on the thread that releases the last
     *                                      reference to the state. In production use cases this should always be true.
     *                                      In unit tests, it should be false unless the test is specifically configured
     *                                      to delete states on a background thread.
```

Reply:
Upon revisiting this topic, I've removed this argument. We will now only delete states on background threads when those states have been instantiated in the regular place in the pipeline. States we load from disk will not be deleted on the background thread.